### PR TITLE
Network Links

### PR DESCRIPTION
**Pull request by FrederoxDev**
As we have discussed privately this PR introduces the concept of "NetworkLinks" into the library. 

## How to use:
To have a block have the ability to be a network link, it must have a few things:
 - It must have the `fluffyalien_energisticscore:network_link` custom component.
 - It must have the `tag:fluffyalien_energisticscore:network_link` block tag.
 - This is all the requirements, things like ticking components are not needed for this type.

## User-facing API:
To create a Network link, we can use the `NetworkLinks.getNetworkLink` function. To this we can pass a `Block` and it will find (or create if it doesn't exist) the network. Under the hood this will instantiate a new entity type which stores all outbound connections to other network links, this entity is completely abstracted away from the end user, since they don't need to care about this.

This function will return a `NetworkLinkNode`, there are two classes that I have added that share this same name, one for the internal core side, and one for the public API. The public API implementation handles sending IPC messages to the backend since we cannot edit the dynamic properties of the data entity from other addons due to dynamic properties being sandboxed. The internals of the core library should never use this public facing implementation of `NetworkLinkNode` since all it does is route IPC messages into the internal version, so the internal version should just be used directly.

I have fully documented the public facing `NetworkLinkNode` class and marked everything with the `@beta` tag.

### CODE REVIEW

File: packs/BP/entities/network_link.json
Comment (Fluffyalien1422) [inline packs/BP/entities/network_link.json:11]: Using a component group and event to despawn this entity is unnecessary. Use [Entity#remove](https://learn.microsoft.com/en-us/minecraft/creator/scriptapi/minecraft/server/entity?view=minecraft-bedrock-stable#remove) instead.

File: packs/BP/scripts/network_links/network_link_component.ts
Comment (Fluffyalien1422) [inline packs/BP/scripts/network_links/network_link_component.ts:41]: A handler is a two-way event, it should not return void. Use [registerScriptEventListener](https://fluffyalien1422.github.io/mcbe-addon-ipc/functions/registerScriptEventListener.html) to register a one-way listener

File: public_api/src/index.ts
Comment (Fluffyalien1422) [inline public_api/src/index.ts:39]: This should be `networkLinks` not `NetworkLinks` as module aliases should be `camelCase`

File: public_api/src/index.ts
Comment (Fluffyalien1422) [inline public_api/src/index.ts:126]: Create a `getNamespace` function instead of exporting this as mutable

**Review by Fluffyalien1422 — CHANGES_REQUESTED** [review]

File: packs/BP/entities/network_link.json
Comment (FrederoxDev) [reply packs/BP/entities/network_link.json:11]: Never knew that existed. Neat.

File: public_api/src/index.ts
Comment (FrederoxDev) [reply public_api/src/index.ts:39]: To me that would indicate that it's an instance of a class or a field for example, when in this case its functionality a namespace

Comment (Fluffyalien1422) [reply public_api/src/index.ts:39]: When you use a module alias in an import you use `camelCase`, it should be the same for an export

File: packs/BP/scripts/network_links/network_link_component.ts
Comment (FrederoxDev) [reply packs/BP/scripts/network_links/network_link_component.ts]: This is required since `dispatchScriptEvent` is not asynchronous, and we need to be able to wait for the operation to be complete due to the delays of script events. Otherwise we could run into order of operation issues where the programmer is expecting the link to have been made successfully at this point, but in actuality it wont be done for a couple ticks.

**Comment by FrederoxDev** [comment]
All the issues should be fixed now, (minus the one with events, which I will wait for your response on, just in case there is another alternative specifically for this case), otherwise should be good to merge if there is nothing else

File: public_api/src/network_links/network_links.ts
Comment (Fluffyalien1422) [inline public_api/src/network_links/network_links.ts:13]: the functions in this file can be turned into static methods on `NetworkLinkNode`, then the `NetworkLinkNode` constructor can be made private.

**Review by Fluffyalien1422 — CHANGES_REQUESTED** [review]

**Review by Fluffyalien1422 — APPROVED** [review]

**Comment by Fluffyalien1422** [comment]
made some changes. waiting for your response before I merge

**Comment by FrederoxDev** [comment]
Lgtm

**Pull request merged**
